src/backend/egl/display.rs

```diff
@@ -103,44 +112,82 @@ unsafe fn select_platform_display<N: EGLNativeDisplay + 'static>(
             continue;
         }
 
-        let display = wrap_egl_call_ptr(|| {
-            ffi::egl::GetPlatformDisplayEXT(
-                platform.platform,
-                platform.native_display,
-                platform.attrib_list.as_ptr(),
-            )
-        })
-        .map_err(Error::DisplayCreationError);
+        let mut hasher = DefaultHasher::new();
+        platform.hash(&mut hasher);
+        let hash = hasher.finish();
```

Comment:
While unlikely couldn't that have collisions resulting in wrongly tracking displays, especially with something like the `DefaultHasher` that could change with a rust release?

From the docs it sounds like `eglGetPlatformDisplay` already tracks this correctly and would return the same handle, so couldn't we just keep calling `eglGetPlatformDisplay` and
test for equality on the returned handle?

Reply:
> While unlikely couldn't that have collisions resulting in wrongly tracking displays, especially with something like the DefaultHasher that could change with a rust release?

Yeah good point. This is the result of me initially wanting to hash the parameters, but not being able to store the `EGLPlatform` inside the `HashMap` due to the native pointer contained.

If we just do a `HashSet<Arc<EGLDisplayHandle>>` instead and keep calling `getPlatformDisplay` I also don't have to change the signature of `select_platform_display`.